website/search/elastic_search.py

```diff
@@ -85,6 +85,9 @@ def wrapped(*args, **kwargs):
                 if 'ParseException' in e.error:
                     raise exceptions.MalformedQueryError(e.error)
                 raise exceptions.SearchException(e.error)
+            except TransportError as e:
+                # Catch and wrap generic ES 500 codes. TODO: Improve fix for https://openscience.atlassian.net/browse/OSF-4538
+                raise exceptions.SearchException(e.error)
 
         sentry.log_message('Elastic search action failed. Is elasticsearch running?')
         raise exceptions.SearchUnavailableError("Failed to connect to elasticsearch")
```

Comment:
While we're here, how about a light refactor to make the logic clearer? This exception could go up under an `if es is None` at the top, and then the `try`/`except` could be dedented a level.